terraform/modules/app/appmesh.tf

```diff
@@ -13,7 +13,7 @@ resource "aws_appmesh_virtual_service" "service" {
 }
 
 resource "aws_appmesh_virtual_node" "service" {
-  count     = length(local.container_services)
+  count     = var.service_mesh ? length(local.container_services) : 0
```

Comment:
It might be worth not using the app module for grafana, since it has a fair few differences. It might make the config for both apps and grafana a bit easier to reason about. What do you think?

Reply:
with you change of introducing the service mesh module, this is not a bit complexity.